sql/src/test/java/io/crate/planner/PlannerTest.java

```diff
@@ -1938,4 +1938,26 @@ public void testQTFPagingIsEnabledOnHighLimit() throws Exception {
         assertThat(plan.localMerge().executionNodes().size(), is(1)); // mergePhase with executionNode = paging enabled
         assertThat(collectPhase.nodePageSizeHint(), is(750000));
     }
+
+    @Test
+    public void testDistributedGroupByProjectionHasShardLevelGranularity() throws Exception {
+        NonDistributedGroupBy nonDistributedGroup = (NonDistributedGroupBy) plan("select count(*) from sys.cluster group by name");
+        CollectPhase collectPhase = nonDistributedGroup.collectPhase();
+        assertThat(nonDistributedGroup, instanceOf(NonDistributedGroupBy.class));
```

Comment:
the result of plan is already casted to NonDistributedGroupBy, so this check is redundant